### PR TITLE
Create the Badge component

### PR DESCRIPTION
**Pull request by wesleycole**
Badge will be used for rendering the count of new items and will accept a label or display only a number.

**Main Story:** [ch59048](https://app.clubhouse.io/skyverge/story/59048/create-the-badge-component-in-kodiak-ui)

<a name="details"></a>

## Details

This PR will address changes to the following:

- [x] Components
- [ ] Hooks

<a name="qa"></a>

## Acceptance Crtieria

- [x] Developers should be able to use the <Badge /> component from Kodiak UI to render a count badge with an optional label

<a name="API Changes"></a>

## API Changes

- [ ] **[Breaking]** This PR introduces breaking changes
- [x] **[Documentation]** Documentation has been written about the API change

<a name="deployment"></a>

### Before merge to master

**Note**: _Code merged to master should be safe to automatically deploy to production as-is._

- [x] **[QA]** User-testing/quality assurance done
- [x] **[Test]** All components and hooks should have a corresponding unit test
- [x] **[Storybook]** All components and hooks should have a corresponding Storybook story
- [x] **[Documentation]** Documentation has been written or updated
- [x] **[Version]** Version number has been updated

### CODE REVIEW

File: packages/primitives/src/Badge/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+type Position = 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right'
+
+function getPositionStyles(position: Position): SxStyleProp {
+  switch (position) {
+    case 'top-right':
+      return {
+        top: 0,
+        right: 0,
+        transform: 'scale(1) translate(50%, -50%)',
```
Comment (alexanderchan) [inline packages/primitives/src/Badge/Badge.tsx:15]: is `scale(1)` for the animation?

File: packages/primitives/src/Badge/Badge.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          color: 'white',
+          display: 'inline-flex',
+          fontSize: 1,
+          height: '20px',
+          justifyContent: 'center',
+          lineHeight: 1,
+          position: hasLabel ? 'static' : 'absolute',
+          py: 0,
+          px: hasLabel ? 2 : '6px',
+          transform: '225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
```
Comment (alexanderchan) [inline packages/primitives/src/Badge/Badge.tsx:107]: should this be a transition for the transform?

File: packages/primitives/src/Badge/Badge.tsx (hunk 1 above)
Comment (wesleycole) [reply packages/primitives/src/Badge/Badge.tsx]: I copied it from Material and I think it is used from going from a dot to a badge with a number. I've removed it fo rnow.

File: packages/primitives/src/Badge/Badge.tsx (hunk 2 above)
Comment (wesleycole) [reply packages/primitives/src/Badge/Badge.tsx]: Transition. I changed it up. Thanks!

File: packages/primitives/src/Badge/Badge.tsx [hunk 3]
```diff
@@ -104,7 +104,7 @@ export const Badge = React.forwardRef(function Badge(
           position: hasLabel ? 'static' : 'absolute',
           py: 0,
           px: hasLabel ? 2 : '6px',
-          transform: '225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
+          transition: '225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
```
Comment (alexanderchan) [inline packages/primitives/src/Badge/Badge.tsx:107]: Hmm I think we need to specify that the transition applies to either `all` or the `transform` but I don't see an animation when we do that 🤷‍♂️ .    If we need to show something without the upper badge, do we just render the component without the wrapper or is there a setting to hide the badge altogether?

Comment (wesleycole) [reply packages/primitives/src/Badge/Badge.tsx:107]: Ah yea I was missing the transform in the transition, thank you!

> If we need to show something without the upper badge, do we just render the component without the wrapper or is there a setting to hide the badge altogether?

What do you mean by this?

Comment (alexanderchan) [reply packages/primitives/src/Badge/Badge.tsx]: I was just wondering how to render the element with no badge:

image

here it's being shown by wrapping in a div but I was wondering if there was an option to pass in to hide the badge altogether.

Comment (wesleycole) [reply packages/primitives/src/Badge/Badge.tsx]: Oh hmm. Material UI has a `showZero` badge otherwise it automatically hides it when at zero.

**Comment by wesleycole** [comment]
@alexanderchan I updated it to add the showZero prop or hide it if count equals zero.

**Pull request merged**